components/dependency_injection/autowiring.rst

```diff
@@ -0,0 +1,411 @@
+.. index::
+    single: DependencyInjection; Autowiring
+
+Using the Autowiring
+====================
+
+Autowiring allows to register services in the container with minimal configuration.
+It is practical in the field of Rapid Application Development, when designing prototypes
+and in early stages of large projects. It makes it easy to bootstrap an app service
+graph and eases refactoring.
+
+Let’s see how it works. To do so we will build a fake API publishing statutes on a Twitter
+feed obfuscated with ROT13 (a special case of the Caesar cipher).
+
+Start by creating a ROT13 transformer class::
+
+    // src/AppBundle/Rot13Transformer.php
+
+    namespace AppBundle;
+
+    class Rot13Transformer
+    {
+        public function transform($value)
+        {
+            return str_rot13($value);
+        }
+    }
+
+And now a Twitter client using this transformer::
+
+    // src/AppBundle/TwitterClient.php
+
+    namespace AppBundle;
+
+    class TwitterClient
+    {
+        private $rot13Transformer;
+
+        public function __construct(Rot13Transformer $rot13Transformer)
+        {
+            $this->rot13Transformer = $rot13Transformer;
+        }
+
+        public function tweetInRot13($user, $key, $status)
+        {
+            $transformedStatus = $this->rot13Transformer->transform($status);
+
+            // Connect to Twitter and send the encoded status
+        }
+    }
+
+
+The Dependency Injection Component is now able to automatically register the dependencies
```

Comment:
The Dependency Injection Component will be able to automatically register the dependencies of this ``TwitterClient`` class by marking the ``twitter_client`` service definition as autowired: